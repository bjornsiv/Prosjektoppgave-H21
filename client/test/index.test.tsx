import * as React from 'react';
import mockReviewService from './mock-review-service';
import mockGameService from './mock-game-service';
import renderer from 'react-test-renderer';
import FrontPage from '../src/index';

jest.mock('../src/game-service', () => {
  return new mockGameService();
});
jest.mock('../src/review-service', () => {
  return new mockReviewService();
});

describe('Game review component tests', () => {
  test('GameReview renders correctly with snapshot', () => {
    const tree = renderer
      .create(<FrontPage/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});