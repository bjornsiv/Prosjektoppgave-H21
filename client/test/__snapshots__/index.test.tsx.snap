// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game review component tests GameReview renders correctly with snapshot 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <h5
        className="card-title"
      >
        GameRatings.com
      </h5>
      <div
        className="card-text"
      >
        Rate top games 
        <button
          className="btn btn-info"
          onClick={[Function]}
          type="button"
        >
          Add game
        </button>
      </div>
    </div>
  </div>
  <div>
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        >
          
        </h5>
        <div
          className="card-text"
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <div
                className="float-start"
              >
                Title
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="float-start"
              >
                Description
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="float-start"
              >
                Genre
              </div>
            </div>
            <div
              className="col"
            >
              <div
                className="float-start"
              >
                Platform
              </div>
            </div>
            <div
              className="col-3"
            >
              <div
                className="float-end"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
